x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/RuntimeKeywordMappedFieldType.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.runtimefields.mapper;
+
+import org.apache.lucene.search.Query;
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.ToXContent.Params;
+import org.elasticsearch.index.fielddata.IndexFieldData;
+import org.elasticsearch.index.mapper.MappedFieldType;
+import org.elasticsearch.index.mapper.TextSearchInfo;
+import org.elasticsearch.index.query.QueryShardContext;
+import org.elasticsearch.script.Script;
+import org.elasticsearch.xpack.runtimefields.StringScriptFieldScript;
+import org.elasticsearch.xpack.runtimefields.fielddata.ScriptBinaryFieldData;
+
+import java.io.IOException;
+import java.util.Map;
+
+public final class RuntimeKeywordMappedFieldType extends MappedFieldType {
+
+    private final Script script;
+    private final StringScriptFieldScript.Factory scriptFactory;
+
+    RuntimeKeywordMappedFieldType(String name, Script script, StringScriptFieldScript.Factory scriptFactory, Map<String, String> meta) {
+        super(name, false, false, TextSearchInfo.NONE, meta);
+        this.script = script;
+        this.scriptFactory = scriptFactory;
+    }
+
+    RuntimeKeywordMappedFieldType(RuntimeKeywordMappedFieldType ref) {
+        super(ref);
+        this.script = ref.script;
+        this.scriptFactory = ref.scriptFactory;
+    }
+
+    @Override
+    public MappedFieldType clone() {
+        return new RuntimeKeywordMappedFieldType(this);
+    }
+
+    @Override
+    public Object valueForDisplay(Object value) {
+        if (value == null) {
+            return null;
+        }
+        // keywords are internally stored as utf8 bytes
+        BytesRef binaryValue = (BytesRef) value;
```

Comment:
I think our keywords come out as strings though, right? When does this get called?

Reply:
no idea, this is copied from the keyword field....